### DISCUSSION TITLE
How to update core packages to unreleased versions?

### DISCUSSION THREAD

**Discussion by HarlemSquirrel** [atom/atom#22184] [Q&A] [answered]
How can I install/update a core package to an unreleased version?

For instance, the [language-ruby package has no been updated since Nov 2019](https://github.com/atom/language-ruby/issues/293) but the master branch includes fixes I would like to have. Running `apm install atom/language-ruby` seems to install the latest tag, not what's in the master branch.

**Comment by ThatXliner** [comment, marked as answer]
`apm install language-ruby@version`. But if you want the master version... I wonder if `@master` would work

**Reply by HarlemSquirrel** [reply]
Nope
```
➤  apm install language-ruby@master
The language-ruby package is bundled with Atom and should not be explicitly installed.
You can run `apm uninstall language-ruby` to uninstall it and then the version bundled
with Atom will be used.
Installing language-ruby@master to /home/kmack/.atom/packages ✗
Package version: master not found
```

**Reply by ThatXliner** [reply]
Dang. What about git cloning the repo, `cd` into the directory, `apm link --dev`.

**Comment by UziTech** [comment]
1. Clone repo
2. Disable core package in Atom
3. In cloned folder run `apm link --name {package name}-temp`
(The name you give to the package must be different from the package name so I usually just append `-temp`)

**Reply by HarlemSquirrel** [reply]
This works but I like using the `--dev` flag instead of a different name since it seems to align with it's intended use.

**Reply by UziTech** [reply]
Ya then you just have to run Atom in development mode in order to use that package.

**Reply by SampsonCrowley** [reply]
@UziTech THANKS! though might I suggest taking a page from the arch linux book and using the postfix `-git` instead of temp. makes it very clear where the package came from

**Reply by SampsonCrowley** [reply]
I'd also suggest just changing the package name in `package.json` instead of using `--name` so that `apm unlink` and the README also works

**Reply by SampsonCrowley** [reply]
actually scratch that, that comes with its own set of issues

**Reply by SampsonCrowley** [reply]
@HarlemSquirrel https://github.com/SampsonCrowley/language-ruby-git is a fork of `atom/language-ruby` implementing https://github.com/atom/language-ruby/pull/291.

if you switch to branch `git-package` then `apm link` will work "as is", including settings management and all, without needing to run dev mode
